Makefile

```diff
@@ -0,0 +1,67 @@
+#! /usr/bin/env make -f
+
+# If we set up the Python virtual environment initially then make sure that
+# the variable 'PYTHON' is available, either its default or passed in by the
+# user. Any other goal requires the Python virtual environment.
+ifeq ($(MAKECMDGOALS),setup)
+  ifndef PYTHON
+    PYTHON=python3.10
+    $(info No Python version specified, defaulting to $(PYTHON))
+  endif
+else
+  ifeq ("$(wildcard .venv)","")
+    $(error No Python environment found, use `make setup` first)
+  else
+    PACKAGE_VERSION=$(shell .venv/bin/python -c 'import package; print(package.__version__)')
+  endif
+endif
+
+.PHONY: setup
+setup:
+	$(PYTHON) -m venv --upgrade-deps .venv
+	. .venv/bin/activate && pip install --upgrade pip && pip install --editable .[dev,test,docs]
+
+.PHONY: all
+all: check test dist
+
+dist: bdist_wheel sdist
+bdist_wheel: .venv/dist/package-$(PACKAGE_VERSION)-py3-none-any.whl
+.venv/dist/package-$(PACKAGE_VERSION)-py3-none-any.whl:
+	. .venv/bin/activate && \
+	python setup.py bdist_wheel --dist-dir .venv/dist/ --bdist-dir .venv/build
+sdist: .venv/dist/package-$(PACKAGE_VERSION).tar.gz
+.venv/dist/package-$(PACKAGE_VERSION).tar.gz:
+	. .venv/bin/activate && \
+	python setup.py sdist --dist-dir .venv/dist/
+
+.PHONY: quick-check check
+quick-check:
+	. .venv/bin/activate && \
+	pre-commit run pylint --all-files && \
+	pre-commit run mypy --all-files
+check:
+	. .venv/bin/activate && \
+	pre-commit run --all-files
+
+.PHONY: test
+test:
+	. .venv/bin/activate && \
+	pre-commit run pytest --hook-stage push
+
+.PHONY: docs
+docs: docs/_build/html/index.html
+docs/_build/html/index.html:
+	$(MAKE) -C docs/ html
+
+.PHONY: dist-clean clean
```

Comment:
I don't think this target requires an activated virtual environment. Especially when things go wrong, we should allow a cleanup nonetheless.

Reply:
Commit https://github.com/jenstroeger/python-package-template/pull/74/commits/90b0a842b050937f9446f96e276e5c854d9e9647.